ArchiSteamFarm.OfficialPlugins.Monitoring/MonitoringPlugin.cs

```diff
@@ -0,0 +1,229 @@
+
+namespace ArchiSteamFarm.OfficialPlugins.Monitoring;
+
+[Export(typeof(IPlugin))]
+[SuppressMessage("ReSharper", "MemberCanBeFileLocal")]
+internal sealed class MonitoringPlugin : OfficialPlugin, IWebServiceProvider, IGitHubPluginUpdates, IASF, IDisposable {
+	private const string MeterName = nameof(ArchiSteamFarm);
+
+	private const string MetricNamePrefix = "asf";
+
+	[JsonInclude]
+	[Required]
+	public override string Name => nameof(MonitoringPlugin);
+
+	/// <inheritdoc />
+	public string RepositoryName => SharedInfo.GithubRepo;
+
+	[JsonInclude]
+	[Required]
+	public override Version Version => typeof(MonitoringPlugin).Assembly.GetName().Version ?? throw new InvalidOperationException(nameof(Version));
+
+	private bool Enabled => (ASF.GlobalConfig?.IPC ?? false) && (Config?.Enabled ?? MonitoringConfig.DefaultEnabled);
+
+	private Meter? Meter;
+
+	[JsonInclude]
+	private MonitoringConfig? Config { get; set; }
+
+	public void Dispose() => Meter?.Dispose();
+
+	public Task OnASFInit(IReadOnlyDictionary<string, JsonElement>? additionalConfigProperties = null) {
+		if (additionalConfigProperties == null) {
+			return Task.CompletedTask;
+		}
+
+		MonitoringConfig? config = null;
+
+		if (additionalConfigProperties.TryGetValue(nameof(Monitoring), out JsonElement configValue)) {
+			try {
+				config = configValue.Deserialize<MonitoringConfig>();
+			} catch (Exception e) {
+				ASF.ArchiLogger.LogGenericException(e);
+				ASF.ArchiLogger.LogGenericWarning(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+				return Task.CompletedTask;
+			}
+		}
+
+		config ??= new MonitoringConfig();
+
+		if (!Enabled) {
+			ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+			return Task.CompletedTask;
+		}
+
+		Config = config;
+
+		ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginInitializedAndEnabled, nameof(MonitoringPlugin)));
+
+		return Task.CompletedTask;
+	}
+
+	public void OnConfiguringEndpoints(IApplicationBuilder app) {
+		ArgumentNullException.ThrowIfNull(app);
+
+		if (!Enabled) {
+			return;
+		}
+
+		app.UseEndpoints(static builder => builder.MapPrometheusScrapingEndpoint());
+	}
+
+	public void OnConfiguringServices(IServiceCollection services) {
+		ArgumentNullException.ThrowIfNull(services);
+
+		if (!Enabled) {
+			return;
+		}
+
+		services.AddOpenTelemetry().WithMetrics(
+			static builder => {
+				builder.AddPrometheusExporter(static config => config.ScrapeEndpointPath = "/Api/metrics");
+				builder.AddMeter("Microsoft.AspNetCore.Hosting");
+				builder.AddMeter("Microsoft.AspNetCore.Server.Kestrel");
+				builder.AddMeter(MeterName);
+			}
+		);
+	}
+
+	public override Task OnLoaded() {
+		Utilities.WarnAboutIncompleteTranslation(Strings.ResourceManager);
+
+		Meter = new Meter(MeterName, Version.ToString());
+
+		_ = Meter.CreateObservableGauge(
```

Comment:
I hate that `_ = ...`, just call `Meter.CreateObservableGauge()` and stop making :clown_face: out of yourself :joy: